x-pack/plugins/apm/server/lib/service_map/ml_helpers.ts

```diff
@@ -0,0 +1,47 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import { SERVICE_NAME } from '../../../common/elasticsearch_fieldnames';
+import {
+  getMlJobServiceName,
+  getSeverity
+} from '../../../common/ml_job_constants';
+import { AnomaliesResponse, ServicesResponse } from './get_service_map';
+
+export function addAnomaliesToServicesData(
+  servicesData: ServicesResponse,
+  anomaliesResponse: AnomaliesResponse
+) {
+  const anomaliesMap = (
+    anomaliesResponse.aggregations?.jobs.buckets ?? []
+  ).reduce<{
+    [key: string]: { max_score?: number };
+  }>((previousValue, currentValue) => {
+    const key = getMlJobServiceName(currentValue.key.toString());
+
+    return {
+      ...previousValue,
+      [key]: {
+        max_score: Math.max(
+          previousValue[key]?.max_score ?? 0,
+          currentValue.max_score.value ?? 0
+        )
```

Comment:
should we distinguish between a actual score of 0 and an unknown score?

Reply:
The things we're reducing over should always have a score, but the type allows for it being undefined, so I think it's ok.